pynio/service.py

```diff
@@ -53,7 +53,12 @@ def connect(self, blk1, blk2=None):
         # if block exists, add the receiever, otherwise init connection
         receivers = [] if blk2 is None else [blk2.name]
         if connection:
-            connection['receivers'].extend(receivers)
+            cr = connection['receivers']
+            cr.extend(receivers)
+            # remove duplicates but preserve order
+            seen = set()
+            seen_add = seen.add
```

Comment:
seen.add is much slower

http://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-in-python-whilst-preserving-order